internal/persister/mongo_persister.go

```diff
@@ -49,6 +50,32 @@ func (mp *MongoPersister) Get(ctx context.Context, id string) (*ResultModel, err
 	return res, nil
 }
 
+func (mp *MongoPersister) GetByTags(ctx context.Context, tags []string) ([]*ResultModel, error) {
+	ctx, cancel := context.WithTimeout(ctx, 5*time.Second)
+	defer cancel()
+	cursor, err := mp.client.Database("toggl_db").
+		Collection("receipt_results").
+		Find(ctx,
+			bson.D{
+				{
+					"tags",
+					bson.D{{"$all", tags}},
+				},
+			})
+	if err != nil {
+		return nil, err
+	}
+	res := make([]*ResultModel, 0)
+	err = cursor.All(ctx, &res)
+	if err != nil {
```

Comment:
opinionated nitpick: I'd use switch to reduce nesting case err == nil: ... case errors.Is(): .... default: ...
I feel switches are easier to read, but the blocks are tiny so it's fine.